lib/routemap.c

```diff
@@ -811,6 +811,17 @@ struct route_map *route_map_lookup_by_name(const char *name)
 	return map;
 }
 
+/* Simple helper to warn if route-map does not exist. */
+struct route_map *route_map_lookup_warn_noexist(struct vty *vty, const char *name)
+{
+	struct route_map *route_map = route_map_lookup_by_name(name);
+
+	if (!route_map)
+		vty_out(vty, "The route-map '%s' does not exist.", name);
```

Comment:
Missing `\n` in the end.

Also, maybe you could print this message only if `vty->type == VTY_TERM` to avoid flooding stderr unnecessarily when reading the startup configuration (when not using the integrated config model).

Reply:
I don't think `\n` is needed here, output is like below:
```
spine1-debian-9(config-router-af)# neighbor 10.0.0.2 route-map noexist out 
The route-map 'noexist' does not exist.
spine1-debian-9(config-router-af)# 
```
Regarding `vty->type == VTY_TERM` as well, it's not printed during startup, only run-time.

Reply:
I believe Renato is correct here, not having a \n will cause the vty terminal to not be moved to the next line.

Reply:
> Regarding vty->type == VTY_TERM as well, it's not printed during startup, only run-time.

Please see the output below:
```
# cat /etc/frr/bgpd.conf
[snip]
!
router bgp 1
 neighbor 10.0.1.1 remote-as 1
 !
 address-family ipv4 unicast
  neighbor 10.0.1.1 route-map not-existing-route-map out
  network 2.2.2.2/32
 exit-address-family
 !
!
route-map not-existing-route-map permit 10
!
#
# bgpd --log=stdout
2018/10/10 13:11:12 BGP: bgpd 6.1-dev starting: vty@2605, bgp@<all>:179
2018/10/10 13:11:12 BGP: Creating Default VRF, AS 1
2018/10/10 13:11:12 BGP: Registering VRF 0
The route-map 'not-existing-route-map' does not exist.
2018/10/10 13:11:12 BGP: Rx Router Id update VRF 0 Id 10.0.1.2/32
```

I think a simple `vty->type == VTY_TERM` check would fix this (except when using the integrated configuration model).

Reply:
Strange, that I didn’t see this in journalctl with info log level. Thanks, I will check this.

Reply:
When reading a configuration file (`vty_read_file()`), the `vty_out()` function outputs to the standard error (`vty->wfd = STDERR_FILENO;`), and not to the standard output nor any log file.

Reply:
Updated. But adding additional check with `VTY_SHELL_SERV` as a vty type. Tested with your given example `bgpd --log=...`, journalctl and inside terminal.